### PR TITLE
[WIP]client: Fix cause of multi-override request crashing

### LINKED ISSUE #3534: override request with multiple package makes client crash
```
[mtasaka@localhost ~]$ rpm -q bodhi-client
bodhi-client-4.1.0-3.fc31.noarch
[mtasaka@localhost ~]$ bodhi overrides save --user mtasaka --duration 4 --notes "ruby-gnome suite 3.3.8 stage 2" "rubygem-gobject-introspection-3.3.8-1.fc30 rubygem-cairo-gobject-3.3.8-1.fc30"
mtasaka's rubygem-gobject-introspection-3.3.8-1.fc30 override (expires 2019-10-08 02:32:48)
mtasaka's rubygem-cairo-gobject-3.3.8-1.fc30 override (expires 2019-10-08 02:32:48)
Caveats:
Your override submission was split into 2.
Traceback (most recent call last):
  File "/usr/lib/python3.7/site-packages/munch/__init__.py", line 102, in __getattr__
    return object.__getattribute__(self, k)
AttributeError: 'Munch' object has no attribute 'build'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.7/site-packages/munch/__init__.py", line 105, in __getattr__
    return self[k]
KeyError: 'build'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/bin/bodhi", line 11, in <module>
    load_entry_point('bodhi-client==4.1.0', 'console_scripts', 'bodhi')()
  File "/usr/lib/python3.7/site-packages/click/core.py", line 763, in __call__
    return self.main(*args, **kwargs)
  File "/usr/lib/python3.7/site-packages/click/core.py", line 716, in main
    rv = self.invoke(ctx)
  File "/usr/lib/python3.7/site-packages/click/core.py", line 1136, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/lib/python3.7/site-packages/click/core.py", line 1136, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/lib/python3.7/site-packages/click/core.py", line 955, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/lib/python3.7/site-packages/click/core.py", line 554, in invoke
    return callback(*args, **kwargs)
  File "/usr/lib/python3.7/site-packages/bodhi/client/__init__.py", line 257, in wrapper
    method(*args, **kwargs)
  File "/usr/lib/python3.7/site-packages/bodhi/client/__init__.py", line 969, in save_buildroot_overrides
    **kwargs)
  File "/usr/lib/python3.7/site-packages/bodhi/client/__init__.py", line 290, in _save_override
    command = _generate_wait_repo_command(resp, client)
  File "/usr/lib/python3.7/site-packages/bodhi/client/__init__.py", line 1013, in _generate_wait_repo_command
    if 'release_id' in override.build:
  File "/usr/lib/python3.7/site-packages/munch/__init__.py", line 107, in __getattr__
    raise AttributeError(k)
AttributeError: build
```

### PR DESCRIPTION
**Pull request by richardgreg**
The _generate_wait_repo_command function is not configured to handle the
creation of multiple overrides. This patch is intended to correct that.

Fixes: #3534

Signed-off-by: Richard O. Gregory <richardgrecoson@gmail.com>

### CODE REVIEW

**Comment by richardgreg** [comment]
I know I have to configure `_save_override` to process the list. I want to present my approach.

File: bodhi/client/__init__.py [hunk 1]
```diff
@@ -1085,13 +1083,44 @@ def _generate_wait_repo_command(override: munch.Munch, client: bindings.BodhiCli
         If we know the release for the override's build, we return a tuple suitable for passing to
         subprocess.Popen for a koji command that will wait on the repo. If we can't we return None.
     """
+    # Creating multiple overrides returns a dict object with keys 'overrides' and 'caveat' having a
+    # length of 2. The response for each override is stored in a list with key 'overrides'  
+    if len(override) == 2:
+        releases = []
+        for i in range(len(override["overrides"])):
```
Comment (cverna) [inline bodhi/client/__init__.py]:
```suggestion
        for i, _ in enumerate(override["overrides"]):
```

**Review by cverna — COMMENTED** [review]
That looks good to me, but we will need some tests before merging :)

**Comment by richardgreg** [comment]
@cverna should I replace `test_create_multiple_overrides` in `test__init__.py` or should I keep it and write additional tests?
Also, are the checks failing on this PR related to my patch?

**Comment by cverna** [comment]
> @cverna should I replace `test_create_multiple_overrides` in `test__init__.py` or should I keep it and write additional tests?
> Also, are the checks failing on this PR related to my patch?
So all the diff-cover tests means that the there some lines that are not covered by a unit test. You can run that command in vagrant `bci diff-cover -r f31` to see which lines it is complaining.

Generally we want to have 100% code coverage for PRs.

**Comment by cverna** [comment]
Closing since there is not activity on this PR

**Pull request closed without merging**